packages/block-editor/src/components/rich-text/index.js

```diff
@@ -243,45 +284,27 @@ export class RichText extends Component {
 	 * @param {PasteEvent} event The paste event.
 	 */
 	onPaste( event ) {
-		const clipboardData = event.clipboardData;
-		let { items, files } = clipboardData;
-
-		// In Edge these properties can be null instead of undefined, so a more
-		// rigorous test is required over using default values.
-		items = isNil( items ) ? [] : items;
-		files = isNil( files ) ? [] : files;
+		const data = extractData( event.clipboardData );
 
-		let plainText = '';
-		let html = '';
-
-		// IE11 only supports `Text` as an argument for `getData` and will
-		// otherwise throw an invalid argument error, so we try the standard
-		// arguments first, then fallback to `Text` if they fail.
-		try {
-			plainText = clipboardData.getData( 'text/plain' );
-			html = clipboardData.getData( 'text/html' );
-		} catch ( error1 ) {
-			try {
-				html = clipboardData.getData( 'Text' );
-			} catch ( error2 ) {
-				// Some browsers like UC Browser paste plain text by default and
-				// don't support clipboardData at all, so allow default
-				// behaviour.
-				return;
-			}
+		if ( ! data ) {
+			return;
 		}
 
 		event.preventDefault();
 
+		this.onProcessData( this.getRecord(), data );
+	}
+
+	onProcessData( record, { plainText, html, files } ) {
 		// Allows us to ask for this information when we get a report.
 		window.console.log( 'Received HTML:\n\n', html );
 		window.console.log( 'Received plain text:\n\n', plainText );
 
 		// Only process file if no HTML is present.
 		// Note: a pasted file may have the URL as plain text.
-		const item = find( [ ...items, ...files ], ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) );
-		if ( item && ! html ) {
-			const file = item.getAsFile ? item.getAsFile() : item;
+		const file = find( files, ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) );
```

Comment:
No need to use Lodash here:

```suggestion
		const file = files.find( ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) );
```

Reply:
Or if `files` may be nullish:

```suggestion
		const file = files?.find( ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) );
```